### PR TITLE
Port to Keras Core complete

### PR DESCRIPTION
**Pull request by sanatsingh**
Made necessary changes for porting to keras_core (TF backend)

### CODE REVIEW

File: .devcontainer/setup.sh [hunk 1]
```diff
@@ -2,3 +2,4 @@ sudo pip install --upgrade pip
 sudo pip install --upgrade tensorflow==2.11.0
 sudo pip install -e .
 sudo pre-commit install
+sudo pip install -q keras-core
```
Comment (soumik12345) [inline .devcontainer/setup.sh:5]: Little curious as to why `sudo` is required!

```suggestion
sudo pip install namex
sudo apt install python3.10-venv
sudo git clone https://github.com/keras-team/keras-core.git && cd keras-core && python pip_build.py --install
```

File: pynet/model/modules.py [hunk 2]
```diff
@@ -46,7 +49,7 @@ def level_4(
     # Expected Output shape for image of 224x224 -> 224/2**(3) -> 28x28x3
     # Input->28x28x256
     # Concat layer
-    l4_cat_1 = tf.concat([l4_out_1, l5_pass_a], -1)
+    l4_cat_1 = keras.layers.Concatenate(axis=-1)([l4_out_1, l5_pass_a])
```
Comment (soumik12345) [inline pynet/model/modules.py:52]: Nice!

File: pynet/model/layers.py [hunk 3]
```diff
@@ -1,5 +1,8 @@
+import os
+os.environ["KERAS_BACKEND"] = "tensorflow"
+
 import tensorflow as tf
-from tensorflow import keras
+import keras_core as keras
 
 import tensorflow_addons as tfa
```
Comment (soumik12345) [inline pynet/model/layers.py:7]: Let's get rid of `tensorflow_addons` as well! We're currently using it for InstanceNorm, which can be replaced with [`keras.layers.GroupNormalization(groups=<number-of-channels-of-input-tensor>)`](https://keras.io/keras_core/api/layers/normalization_layers/group_normalization/).

**Review by soumik12345 — CHANGES_REQUESTED** [review]
Hi @sanatsingh 
Great job on the PR!
Left some feedback...

File: .devcontainer/setup.sh (hunk 1 above)
Comment (sanatsingh) [reply .devcontainer/setup.sh:5]: Not sure about sudo but do we want to install keras-core directly from git repo or stick to stable releases based on versions?

File: pynet/model/layers.py (hunk 3 above)
Comment (sanatsingh) [reply pynet/model/layers.py:7]: Done

File: .devcontainer/setup.sh (hunk 1 above)
Comment (sanatsingh) [reply .devcontainer/setup.sh]: Updated installation command

**Review by soumik12345 — APPROVED** [review]
